testing/integration/beat_receivers_test.go

```diff
@@ -201,185 +238,225 @@ func TestClassicAndReceiverAgentMonitoring(t *testing.T) {
-					key := tc.dsType + "-" + tc.dsDataset + "-" + tc.dsNamespace
-					otelDocs[key] = docs
-				}
-				return docs.Hits.Total.Value > 0
-			},
-			4*time.Minute, 5*time.Second,
-			"agent monitoring beats receivers no documents found for timestamp: %s, type: %s, dataset: %s, namespace: %s, query: %v", timestampBeatReceiver, tc.dsType, tc.dsDataset, tc.dsNamespace, tc.query)
-	}
+						"sort": []map[string]any{
+							{"@timestamp": map[string]any{"order": "asc"}},
+						},
+					}
 
-	// 6. Uninstall
-	combinedOutput, err = beatReceiverFixture.Uninstall(ctx, &atesting.UninstallOpts{Force: true})
-	require.NoErrorf(t, err, "error uninstalling beat receiver agent monitoring, err: %s, combined output: %s", err, string(combinedOutput))
+					docs, err := estools.PerformQueryForRawQuery(findCtx, rawQuery, tc.dsType+"-*", info.ESClient)
+					require.NoError(t, err)
+					if docs.Hits.Total.Value != 0 {
+						key := tc.dsType + "-" + tc.dsDataset + "-" + processNamespace
+						agentDocs[key] = docs
+					}
+					return docs.Hits.Total.Value > 0
+				},
+				5*time.Minute, 5*time.Second,
+				"process based agent monitoring no documents found for timestamp: %s, type: %s, dataset: %s, namespace: %s, query: %v", processTimestamp, tc.dsType, tc.dsDataset, processNamespace, tc.query)
+		}
+		// Uninstall process based elastic-agent monitoring
+		combinedOutput, err := processFixture.Uninstall(ctx, &atesting.UninstallOpts{Force: true})
+		require.NoErrorf(t, err, "error uninstalling process based agent monitoring, err: %s, combined output: %s", err, string(combinedOutput))
+
+		// beats processes and beats receivers need different namespaces
+		receiverNamespace := fmt.Sprintf("%s-%s", info.Namespace, "otel")
+		policy.Agent.Monitoring["namespace"] = receiverNamespace
+
+		// switch monitoring to the otel runtime
+		policy.Agent.Monitoring["_runtime_experimental"] = "otel"
+
+		receiverPolicyBytes, err := yaml.Marshal(policy)
+		require.NoErrorf(t, err, "error marshalling policy, struct was %v", policy)
+
+		// Install without enrolling in fleet
+		receiverFixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+
+		err = receiverFixture.Prepare(ctx)
+		require.NoError(t, err, "error preparing receiver agent monitoring fixture")
+
+		err = receiverFixture.Configure(ctx, receiverPolicyBytes)
+		require.NoError(t, err, "error configuring receiver agent monitoring fixture")
+
+		output, err = receiverFixture.InstallWithoutEnroll(ctx, &installOpts)
+		require.NoErrorf(t, err, "error install with beat receiver monitoring without enroll: %s\ncombinedoutput:\n%s", err, string(output))
+		timestampBeatReceiver := time.Now().UTC().Format("2006-01-02T15:04:05.000Z")
+
+		require.EventuallyWithT(t, func(collect *assert.CollectT) {
+			status, statusErr := receiverFixture.ExecStatus(ctx)
+			assert.NoError(collect, statusErr)
+			// agent should be healthy
+			assert.Equal(collect, int(cproto.State_HEALTHY), status.State)
```

Comment:
You might want to apply these comments as the last argument of the assert or require functions. It will turn failure messages from something like "1 is not equal to 0" to "1 is not equal to 0 [...] agent status should be healthy" which is very helpful for debugging.